src/artemis/stores/mapgraph/write.cljs

```diff
@@ -107,12 +106,14 @@
 
                                   nsed-key (str stub "." (name sel-key))
                                   new-v (if (sequential? v)
-                                          (map #(format-for-cache context (:selection-set sel) %1 nsed-key %2) v (range))
+                                          (mapv (fn [result idx]
+                                                  (format-for-cache context (:selection-set sel) result (str nsed-key "." idx)))
```

Comment:
`(str nsed-key "." idx)` is the fix, along with removing the other index tracking.

Reply:
Nice spelunking